### PR TITLE
Integration spec

### PR DESCRIPTION
**Pull request by mosidrum**
In this PR, I did the following:

- [ ] Use Capybara to write integration tests for each view in my project.

- [ ] I wrote test for checking for the things in index page, user show page, User post index page and  for clicking links on each page.

- [ ] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user

**Screenshot before fixing the problem**

**Screenshot after fixing the problem**

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    visit user_posts_path(@first_user)
+    expect(page).to have_content(@second_post.likes_counter)
+  end
+
+  it 'Click a user\'s post, redirects to post\'s show page' do
+    visit user_path(@first_user)
+    expect(page).to have_link(@second_post.title)
+    click_link @second_post.title
+    expect(current_path).to eq(user_post_path(@first_user, @second_post))
+  end
```
Comment (rloterh) [inline spec/features/post_index_spec.rb:63]: - [ ] A test to assert the scenario **_I can see a section for pagination if there are more posts than fit on the view_** for _post index_ view integration spec is missing! Please ensure to improve upon this aspect.

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    visit user_post_path(@first_user, @post)
+    expect(page).to have_content(@second_user.name)
+  end
+
+  it 'sees the comment of each commenter' do
+    visit user_post_path(@first_user, @post)
+    expect(page).to have_content(@comment.text)
+  end
+end
+
```
Comment (rloterh) [inline spec/features/post_show_spec.rb]: - [ ] A test to assert the scenario **_I can see how many comments it has_** for _post show_ view integration spec is missing! Please ensure to improve upon this aspect.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Isaac, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 
🏁 🚦  ⛪ 🚜  🏃   🌴  🏫 🌳 🏘️ 

## Highlights 👍  
- N + 1 problem addressed!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+end
+
```
Comment (DammyShittu) [inline spec/features/post_index_spec.rb]: - [ ] Kindly add the pagination requirement from the previous reviewer. We need to test that there is pagination for the scenario when there are too many posts to fit the page.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @mosidrum and @rodrigopenayo98,

## Highlights
✔️ Passing tests

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi @supreeti 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
AmazingCreepyFrogmouth-max-1mb




## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅


## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**